dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolDescription.java

```diff
@@ -0,0 +1,131 @@
+    private final AnnotationsDescription annotations;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public InputSchemaObject getInputSchema() {
+        return inputSchema;
+    }
+
+    public AnnotationsDescription getAnnotations() {
+        return annotations;
+    }
+
+    public ToolDescription(ToolMetadata toolMetadata) {
+        this.name = toolMetadata.name();
+        this.title = toolMetadata.title();
+        this.description = toolMetadata.description();
+
+        Tool.Annotations ann = toolMetadata.annotation().annotations();
+        if (isDefaultAnnotation(ann)) {
+            this.annotations = null;
+        } else {
+            this.annotations = new AnnotationsDescription(
+                                                          ann.readOnlyHint() == false ? null : ann.readOnlyHint(),
+                                                          ann.destructiveHint() == true ? null : ann.destructiveHint(),
+                                                          ann.idempotentHint() == false ? null : ann.idempotentHint(),
+                                                          ann.openWorldHint() == true ? null : ann.openWorldHint(),
+                                                          ann.title().isEmpty() ? null : ann.title());
+        }
+        Map<String, ArgumentMetadata> argumentMap = toolMetadata.arguments();
+        Map<String, InputSchemaPrimitive> primitiveInputSchemaMap = new HashMap<>();
+        LinkedList<String> requiredParameterList = new LinkedList<>();
+
+        for (String argumentName : argumentMap.keySet()) {
+            primitiveInputSchemaMap.put(argumentName, buildPrimitiveInputSchema(argumentMap.get(argumentName)));
+            requiredParameterList.add(argumentName);
+        }
+
+        inputSchema = new InputSchemaObject("object", primitiveInputSchemaMap, requiredParameterList);
+    }
+
+    /*
+     * Helper Method for default Annotation
+     */
+
+    private boolean isDefaultAnnotation(Tool.Annotations ann) {
+        return ann.readOnlyHint() == false
+               && ann.destructiveHint() == true
+               && ann.idempotentHint() == false
+               && ann.openWorldHint() == true
+               && ann.title().isEmpty();
+
+    }
+
+    private InputSchemaPrimitive buildPrimitiveInputSchema(ArgumentMetadata argumentMetadata) {
+
+        Type type = argumentMetadata.type();
+        String argumentDescription = argumentMetadata.description();
+        if (argumentDescription.equals("")) {
+            argumentDescription = null;
+        }
+
+        InputSchemaPrimitive tempSchemaPrimitive;
+
+        boolean isJsonNumber = type.equals(long.class) || type.equals(double.class) || type.equals(byte.class) || type.equals(float.class) || type.equals(short.class)
```

Comment:
Probably would be easier to do with a static Set that you just do a contains passing in type instead of all of these ORs and equals.  Or having a map that returns an impl of a Converter interface that returns the values.  If one isn't in the map, then it just does the final else.

Reply:
This is all early prototype code. I expect this part will end up being replaced in #32572 when we extend schema support.